File: C/04_expressions_and_operators/src/bool_exp.c
#include <stdio.h>

int main(int argc, char **argv)
{
  float x = 42;

  printf("x = %.1f\n", x);
  if (0 <= x <= 1) {
    printf("whoa!  0 <= %.1f <= 1 evaluates to true!\n", x);
  }

  x = -1.0;
  printf("x = %.1f\n", x);
  if ((0 <= x) <= 0.5) {
    printf("ok!  (0 <= %.1f) <= 0.5 evaluates to true!\n", x);
  }

  return 0;
}

File: C/04_expressions_and_operators/src/bool_exp.py
x = 42.0
print(f"{x = }")
if 0 <= x <= 1:
    print(f"whoa!  0 <= {x} <= 1 evaluates to True!")
else:
    print(f"yay!  0 <= {x} <= 1 evaluates to False!")

x = -1
print(f"{x = }")
if (0 <= x) <= 1:
    print(f"whoa!  (0 <= {x}) <= 1 evaluates to True!")
else:
    print(f"yay!  0 <= {x} <= 1 evaluates to False!")
